inc/sitemaps/class-post-type-sitemap-provider.php

```diff
@@ -283,6 +283,8 @@ public function save_post( $post_id ) {
 	 */
 	public function is_valid_post_type( $post_type ) {
 
+		global $wp_post_types;
```

Comment:
This might be called significant amount of times in different contexts. Using API call might be slow and caching it might make it miss post types because of registration timing.

Global access is a compromise to have reasonably fast access to a most current state.